### DISCUSSION TITLE
Docker: where is located PMS?

### DISCUSSION THREAD

**Discussion by Maelep** [UniversalMediaServer/UniversalMediaServer#2739] [Q&A] [answered]
I'm trying to write a docker image based on yours but I don't understand `-cp ums.jar net.pms.PMS`.
Where is located `net.pms.PMS`? I don't find it anywhere in the folder your copy.
I made a copy of your `/usr/src/ums` that you copy in your docker image but `net.pms.PMS` isn't located there.

**Comment by SubJunk** [comment]
@Maelep `net.pms.PMS` is a reference to the main Java class, it's not a file or folder. `-cp` stands for `classPath` and is documented at https://docs.oracle.com/javase/7/docs/technotes/tools/windows/java.html

**Reply by Maelep** [reply]
Thank you for your answer.
So it should be referenced at the location of `ums.jar`?

In Ubuntu its something like `export CLASSPATH=location/of/ums.jar`

**Reply by SubJunk** [reply]
I'm not sure I understand the question, but here's the script we use to launch UMS on Linux in case that helps: https://github.com/UniversalMediaServer/UniversalMediaServer/blob/master/src/main/external-resources/UMS.sh

**Reply by Maelep** [reply]
I'm struggling to starts my docker image because my files are located in `mnt/server/`.

My dockerfile looks like
```dockerfile
FROM universalmediaserver/ums as builder
FROM alpine:3.15.0

COPY --from=builder /usr/src/ums /mnt/server/
```

And using your dockerfile startup
```sh
java $JVM_OPTS -XX:+UnlockExperimentalVMOptions -XX:+UseCGroupMemoryLimitForHeap \
	-DUMS_PROFILE=/profile -Dfile.encoding=UTF-8 -Djava.net.preferIPv4Stack=true -Djna.nosys=true \
	-cp ums.jar net.pms.PMS`
```
Doesn't work because it doesn't find `-cp ums.jar net.pms.PMS`

**Reply by SubJunk** [reply]
Did you see our `dockerfile`? https://github.com/UniversalMediaServer/UniversalMediaServer/blob/master/docker/Dockerfile

**Reply by Maelep** [reply]
Yes, I made mine from yours.
I'm trying to develop some features to add that's why I'm mounting my volumes.

**Reply by SubJunk** [reply]
If it isn't finding the classpath in the `.jar` file, it might mean it's not the right jar file, like it hasn't been built correctly

**Reply by Maelep** [reply]
Thank you for your answer. I managed to make it runs (I wasn't using the good release...).
By the way, I get some error when trying to read YouTube channels:
```java
INFO  03:59:08.078 [Thread-270] Exception in thread "Thread-270" java.lang.NullPointerException
INFO  03:59:08.078 [Thread-270]         at net.pms.remote.RemoteUtil.lambda$dump$0(RemoteUtil.java:106)
INFO  03:59:08.078 [Thread-270]         at java.lang.Thread.run(Thread.java:748)
INFO  04:00:13.366 [StopPlaying Event] Stopped playing NBC's Newest High-Stakes Thriller Is Coming Soon | NBC's The Endgame on Firefox
ERROR 04:00:21.468 [Hanging External Process Stopper] External input stream instance is null... stopping process
INFO  04:00:21.475 [StartPlaying Event] Started playing NBC's Newest High-Stakes Thriller Is Coming Soon | NBC's The Endgame on your Firefox
INFO  04:00:21.499 [Thread-298] Exception in thread "Thread-298" java.lang.NullPointerException
INFO  04:00:21.517 [Thread-298]         at net.pms.remote.RemoteUtil.lambda$dump$0(RemoteUtil.java:106)
INFO  04:00:21.517 [Thread-298]         at java.lang.Thread.run(Thread.java:748)
INFO  04:00:25.856 [StopPlaying Event] Stopped playing NBC's Newest High-Stakes Thriller Is Coming Soon | NBC's The Endgame on Firefox
ERROR 04:00:33.042 [Hanging External Process Stopper] External input stream instance is null... stopping process
INFO  04:00:33.047 [Thread-304] Exception in thread "Thread-304" java.lang.NullPointerException
INFO  04:00:33.048 [Thread-304]         at net.pms.remote.RemoteUtil.lambda$dump$0(RemoteUtil.java:106)
INFO  04:00:33.048 [Thread-304]         at java.lang.Thread.run(Thread.java:748)
INFO  04:00:33.053 [StartPlaying Event] Started playing NBC's Newest High-Stakes Thriller Is Coming Soon | NBC's The Endgame on your Firefox
ERROR 04:00:55.084 [Hanging External Process Stopper] External input stream instance is null... stopping process
INFO  04:00:55.094 [StartPlaying Event] Started playing The Onion Reviews 'Licorice Pizza' on your Firefox
INFO  04:00:55.094 [Thread-394] Exception in thread "Thread-394" java.lang.NullPointerException
INFO  04:00:55.095 [Thread-394]         at net.pms.remote.RemoteUtil.lambda$dump$0(RemoteUtil.java:106)
INFO  04:00:55.095 [Thread-394]         at java.lang.Thread.run(Thread.java:748)
INFO  04:00:59.094 [StopPlaying Event] Stopped playing NBC's Newest High-Stakes Thriller Is Coming Soon | NBC's The Endgame on Firefox
INFO  04:01:00.089 [Thread-395] Exception in thread "Thread-395" java.lang.NullPointerException
INFO  04:01:00.090 [Thread-395]         at net.pms.remote.RemoteUtil.lambda$dump$0(RemoteUtil.java:106)
INFO  04:01:00.090 [Thread-395]         at java.lang.Thread.run(Thread.java:748)
```

**Reply by SubJunk** [reply]
I'm glad it's working now :) if you create a new issue following the instructions for the YouTube problem I can look into it https://github.com/UniversalMediaServer/UniversalMediaServer/issues/new

**Comment by Maelep** [comment]
I'll make more tests before opening an issue, perhaps I made an error in my configuration.
Thank you for your answers.

**Comment by SubJunk** [comment, marked as answer]
For future reference, this problem was caused by the wrong `.jar` file being used: https://github.com/UniversalMediaServer/UniversalMediaServer/discussions/2739#discussioncomment-1861318
